pyomo/contrib/mindtpy/MindtPy.py

```diff
@@ -115,11 +115,15 @@ def solve(self, model, **kwds):
 
             MindtPy = solve_data.working_model.MindtPy_utils
             setup_results_object(solve_data, config)
+            # if ROA is activated, linear objective should be replace by epigraph constraint.
+            # Since the distance calculate do not include these variable, these variables should not be added to variable_list
+            # TODO: check why constraint_list are not updated.
```

Comment:
Should this TODO be left behind on this PR or solved here?

Reply:
This TODO will be solved here.

Reply:
Alright, so let's wait for it to be solved before merging. I don't know why should it fail there though... I'll think about it and see if I can find a solution

Reply:
It might not be a bug. I just forget the implementation details and I'm trying to recall them.

Reply:
This is the only part holding this PR, there is some grammar mistake in the comment `Since the distance calculated does not include these variables, ...` but more importantly, is it not yet clear if the problem has been solved or not.

Reply:
This problem has been resolved. The original implementation is correct and I added more descriptions for it.